dotCMS/src/main/java/com/dotcms/storage/DataBaseStoragePersistenceAPIImpl.java

```diff
@@ -0,0 +1,721 @@
+        final boolean isExternalPool = UtilMethods.isSet(jdbcPool);
+
+        if (isExternalPool) {
+
+            wrapExternalCloseConnection(voidDelegate, jdbcPool);
+        } else {
+
+            wrapLocalCloseConnection(voidDelegate);
+        }
+    }
+
+    /**
+     * custom close connection wrapper
+     * @param voidDelegate
+     */
+    private void wrapLocalCloseConnection(final VoidDelegate voidDelegate) {
+
+        final boolean isNewConnection = !DbConnectionFactory.connectionExists();
+
+        try {
+
+            voidDelegate.execute();
+        } catch (DotSecurityException | DotDataException e) {
+
+            Logger.error(this, e.getMessage(), e);
+            throw new DotRuntimeException(e);
+        } finally {
+
+            if (isNewConnection) {
+
+                DbConnectionFactory.closeSilently();
+            }
+        }
+    }
+
+    /**
+     * custom external close connection wrapper
+     * @param voidDelegate
+     * @param jdbcPool
+     */
+    private void wrapExternalCloseConnection(final VoidDelegate voidDelegate,
+            final String jdbcPool) {
+
+        Connection connection = null;
+        try {
+
+            connection = DbConnectionFactory.getConnection(jdbcPool);
+            voidDelegate.execute();
+        } catch (DotSecurityException | DotDataException e) {
+
+            Logger.error(this, e.getMessage(), e);
+            throw new DotRuntimeException(e);
+        } finally {
+
+            CloseUtils.closeQuietly(connection);
+        }
+    }
+
+    /**
+     * custom transaction wrapper
+     * @param delegate
+     * @param <T>
+     * @return
+     */
+    protected <T> T wrapInTransaction(final ReturnableDelegate<T> delegate) {
+
+        final String jdbcPool = Config
+                .getStringProperty("DATABASE_STORAGE_JDBC_POOL_NAME", StringPool.BLANK);
+        final boolean isExternalPool = UtilMethods.isSet(jdbcPool);
+
+        if (isExternalPool) {
+
+            return this.wrapInExternalTransaction(delegate, jdbcPool);
+        } else {
+
+            try {
+
+                return LocalTransaction.wrapReturnWithListeners(delegate);
+            } catch (Exception e) {
+
+                Logger.error(this, e.getMessage(), e);
+                throw new DotRuntimeException(e);
+            }
+        }
+    }
+
+    /**
+     * custom transaction wrapper
+     * @param delegate
+     * @param jdbcPool
+     * @param <T>
+     * @return
+     */
+    private <T> T wrapInExternalTransaction(final ReturnableDelegate<T> delegate,
+            final String jdbcPool) {
+
+        T result = null;
+        Connection connection = null;
+        boolean autocommit = false;
+
+        try {
+
+            connection = DbConnectionFactory.getConnection(jdbcPool);
+            autocommit = connection.getAutoCommit();
+            connection.setAutoCommit(false);
+
+            result = delegate.execute();
+
+            connection.commit();
+        } catch (Throwable e) {
```

Comment:
Codacy found an issue: [A catch statement should never catch throwable since it includes errors.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)